infrastructure/serve/aml.tf

```diff
@@ -0,0 +1,71 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+resource "local_file" "aml_registry_config" {
+  content  = <<EOF
+name: ${local.aml_registry_name}
+description: Basic AML registry located in ${var.core_rg_location}
+location: ${var.core_rg_location}
+replication_locations:
+  - location: ${var.core_rg_location}
+EOF
+  filename = "registry.yml"
+}
+
+# AML registry usees public network accesss, should be altered when it support private
+resource "null_resource" "az_cli_registry_create" {
+  provisioner "local-exec" {
+    when    = create
+    command = "az extension add --name ml && az ml registry create --file ${local_file.aml_registry_config.filename} --resource-group ${var.core_rg_name} --public-network-access Enabled"
```

Comment:
I know, installing the extension here is less than ideal. Currently the devcontainer.json way of installing the `ml` extension doesn't seem to get picked up - nor does the `onCreateCommand` (in the build) either.

Will have another look and if not log an issue to fix up later.

Reply:
https://github.com/UCLH-Foundry/FlowEHR/issues/286

Reply:
There is another way we could enable this - we could add `az config set extension.use_dynamic_install=yes_without_prompt` as a dev container post create command and then whenever a cli command is invoked that requires an extension (like above) it will install automatically